publication/models/account_analytic_invoice_line.py

```diff
@@ -28,3 +28,62 @@ def action_distribution_list(self):
         action['view_mode'] = 'form'
         action['target'] = 'current'
         return action
+
+    @api.model
+    def create(self, vals):
+        distribution_list_model = self.env['distribution.list']
+        if vals.get('publication'):
+            contract_partner_id = self.env['account.analytic.account'].browse(
+                vals.get('analytic_account_id')).partner_id.id
+            distribution_list_model.create({
+                'product_id': vals.get('product_id'),
+                'contract_partner_id': contract_partner_id,
+                'copies': result.quantity,
+                'partner_id': vals.get('partner_id'),
+                'date_start': vals.get('date_start'),
+            })
+        return super(AccountAnalyticInvoiceLine, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).write(vals)
+        distribution_list = self.env['distribution.list']
+        product_product = self.env['product.product']
+        res_partner = self.env['res.partner']
+        for rec in self:
+            publication = vals.get('publication', None) or self.publication
```

Comment:
why look at vals, if everything is already in the records? Anyway publication will not be in vals.